### PR TITLE
Add Avalanche C-Chain and xDAI

### PR DESCRIPTION
**Pull request by hewigovens**
- Extract C-Chain code from https://github.com/trustwallet/wallet-core/pull/1480
- Add xDAI back

### CODE REVIEW

File: registry.json [hunk 1]
```diff
@@ -1668,5 +1668,53 @@
...
+    "publicKeyType": "secp256k1Extended",
+    "explorer": {
+      "url": "https://blockscout.com/xdai/mainnet",
+      "txPath": "/tx/",
+      "accountPath": "/address/",
+      "sampleTx": "0xcf5162062379ccdb0d4fe9595ff61a2ea8b6a59c79275002e690f0c6ddfda8b4",
+      "sampleAccount": "0x12d61a95CF55e18D267C2F1AA67d8e42ae1368f8"
+  },
+   "info": {
+    "url": "https://www.xdaichain.com",
```
Comment (vikmeup) [inline registry.json:1714]: Can you format json?

**Review by vikmeup — APPROVED** [review]

File: tests/xDai/TWCoinTypeTests.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    auto accUrl = WRAPS(TWCoinTypeConfigurationGetAccountURL(TWCoinTypeXDai, accId.get()));
+    auto id = WRAPS(TWCoinTypeConfigurationGetID(TWCoinTypeXDai));
+    auto name = WRAPS(TWCoinTypeConfigurationGetName(TWCoinTypeXDai));
+
+    ASSERT_EQ(TWCoinTypeConfigurationGetDecimals(TWCoinTypeXDai), 18);
+    ASSERT_EQ(TWBlockchainEthereum, TWCoinTypeBlockchain(TWCoinTypeXDai));
+    ASSERT_EQ(0x0, TWCoinTypeP2shPrefix(TWCoinTypeXDai));
+    ASSERT_EQ(0x0, TWCoinTypeStaticPrefix(TWCoinTypeXDai));
+    assertStringsEqual(symbol, "xDAI");
+    assertStringsEqual(txUrl, "https://blockscout.com/xdai/mainnet/tx/0x9243890b844219accefd8798271052f5a056453ec18984a56e81c92921330d54");
```
Comment (catenocrypt) [inline tests/xDai/TWCoinTypeTests.cpp:30]: Note: this URL is not a valid tx explorer page, not found
https://blockscout.com/xdai/mainnet/tx/0x9243890b844219accefd8798271052f5a056453ec18984a56e81c92921330d54

**Review by catenocrypt — APPROVED** [review]
It's not clear if xDai was present before (I did not find it in recent history), and if it was removed, why.

File: tests/xDai/TWCoinTypeTests.cpp (hunk 2 above)
Comment (hewigovens) [reply tests/xDai/TWCoinTypeTests.cpp]: good catch

**Comment by hewigovens** [comment]
@catenocrypt we removed it before `derivationPath` refactoring and now it uses same path as Ethereum

**Pull request merged**